torch/csrc/lazy/core/hash.h

```diff
@@ -84,7 +84,41 @@ static inline hash_t Hash(const c10::Scalar& value) {
   case c10::ScalarType::Bool:
     return Hash(value.toBool());
   default:
-    TORCH_INTERNAL_ASSERT(false, "Unknown scalar type.");
+    TORCH_INTERNAL_ASSERT(false, "Unknown scalar type.", value.type());
+  }
+}
+
+static inline torch::lazy::hash_t TensorHash(const at::Tensor& tensor) {
+  at::Tensor ctensor = tensor.contiguous();
+  int64_t size = ctensor.numel() * ctensor.element_size();
+  switch (ctensor.scalar_type()) {
+    case at::ScalarType::Bool:
+      return DataHash(ctensor.data_ptr<bool>(), size);
```

Comment:
this seems needlessly complicated doesn't it?  wonder why we can't just 
1) assert scalar_type() is supported (maybe not necessary, or maybe use c10 util for it)
2) treat data_ptr as uint8_t* and call DataHAsh

not sure why it was necessary to handle all the cases separately.. but maybe it's better not to change it now...